riptide-httpclient/src/main/java/org/zalando/riptide/httpclient/RestAsyncClientHttpResponse.java

```diff
@@ -0,0 +1,84 @@
+package org.zalando.riptide.httpclient;
+
+/*
+ * ⁣​
+ * Riptide: HTTP Client
+ * ⁣⁣
+ * Copyright (C) 2015 - 2016 Zalando SE
+ * ⁣⁣
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ​⁣
+ */
+
+import lombok.SneakyThrows;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.client.ClientHttpResponse;
+
+import java.io.Closeable;
+import java.io.FilterInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Field;
+
+final class RestAsyncClientHttpResponse implements ClientHttpResponse {
+
+    private final ClientHttpResponse response;
+
+    RestAsyncClientHttpResponse(final ClientHttpResponse response) {
+        this.response = response;
+    }
+
+    @Override
+    public HttpStatus getStatusCode() throws IOException {
+        return response.getStatusCode();
+    }
+
+    @Override
+    public int getRawStatusCode() throws IOException {
+        return response.getRawStatusCode();
+    }
+
+    @Override
+    public String getStatusText() throws IOException {
+        return response.getStatusText();
+    }
+
+    @Override
+    public InputStream getBody() throws IOException {
+        return new FilterInputStream(response.getBody()) {
+            @Override
+            public void close() throws IOException {
+                // intentionally left blank
+                // Some implementations (looking at you, org.apache.http.impl.io.ChunkedInputStream) actually try
+                // to completely consume the content upon close which of course doesn't work for endless streams.
+                // This will cause the connection to not be released back to the connection pool.
```

Comment:
I think this implementation is wrong. I have debugged this situation and found out, that the connection is still pending until it times out. To finish without reading, the correct action is to implement it as follows:

```
if (in instanceof EofSensorInputStream) {
      ((EofSensorInputStream) in).abortConnection();
} else {
       in.close();
}
```